### PR TITLE
Modernized TrackerRecoGeometryESProducer

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- Added consumes and ESGetTokens
- Added fillDescriptions
- Combined header and source files

#### PR validation:

Code compiles.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: RecoTracker/GeometryESProducer/plugins/TrackerRecoGeometryESProducer.cc [hunk 1]
```diff
@@ -14,32 +14,47 @@
...
   GeometricSearchTrackerBuilder builder;
-  return std::unique_ptr<GeometricSearchTracker>(builder.build(tG->trackerDet(), &(*tG), tTopo));
+  return std::unique_ptr<GeometricSearchTracker>(builder.build(tG.trackerDet(), &tG, &iRecord.get(tTopToken_)));
+}
+
+void TrackerRecoGeometryESProducer::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.addUntracked<std::string>("trackerGeometryLabel", "");
+  descriptions.addDefault(desc);
```
Comment (Dr15Jones) [inline RecoTracker/GeometryESProducer/plugins/TrackerRecoGeometryESProducer.cc]: @perrotta I do not believe this is the same case as in #27589. Here I just changed accessing an untracked parameter with a default to injecting the default via a `fillDescriptions`. To the framework and any outside observer (or even this module) the two ways of doing it are identical. The only difference is now `edmPluginHelp` will actually work.

File: RecoTracker/GeometryESProducer/plugins/TrackerRecoGeometryESProducer.cc [hunk 2]
```diff
@@ -1,7 +1,7 @@
-#include "RecoTracker/GeometryESProducer/plugins/TrackerRecoGeometryESProducer.h"
```
Comment (perrotta) [inline RecoTracker/GeometryESProducer/plugins/TrackerRecoGeometryESProducer.cc]: Shouldn't be added also the headers for
```C++
#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
#include "FWCore/ParameterSet/interface/ParameterSet.h"
```

Comment (Dr15Jones) [reply RecoTracker/GeometryESProducer/plugins/TrackerRecoGeometryESProducer.cc]: done

**Comment by Dr15Jones** [comment]
Please test

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Comment by makortel** [comment]
part of #26748

**Pull request merged**